website/src/blogs/ha-vs-dr-and-ha-c2-b2-for-your-db.md

```diff
@@ -0,0 +1,99 @@
+---
+title: HA vs. DR and HA² for your DB
+author: Evan Powell
+author_info: Founding CEO of a few companies including StackStorm (BRCD) and Nexenta — and CEO & Chairman of OpenEBS/MayaData. ML and DevOps and Python, oh my!
+date: 21-02-2019
+tags: Cloud Storage, DevOps, Database Administration, OpenEBS, Kubernetes
+excerpt: This post is intended to share observations gathered from a couple of decades worth of experience in helping enterprises build resilient systems and presented based on feedback from Kubernetes and StackStorm users over the past 4-5 years.
+---
+
+*Steps to prevent cascading failures.*
+
+This post is intended to share observations gathered from a couple of decades worth of experience in helping enterprises build resilient systems and presented based on feedback from Kubernetes and StackStorm users over the past 4-5 years. As such, mileage may vary; I myself am still learning and offer this as a way for us all to learn together, so feedback is not just welcome, it is absolutely requested.
+
+Here is what I have seen too often, and is the basis of all the principles I share below. For all of us, as we build these systems of systems with more dependencies and dynamic relationships than any one human could possible fully understand, we want to make sure that we don’t generate opaque cascading failures.
+
+In short, Don’t Injure Yourself (DIY)
+
+For starters, don’t make your automation so intelligent that it knows how to kill nodes that are not responding without also understanding *why* that node might be moving slowly. As I have seen before, it could be that  your load peaked the day after Thanksgiving, and by pulling the slow nodes out of the queue you are simply shortening the time before all other nodes become overwhelmed. These are the things brownouts are made of.
+
+So, how can you avoid being thrown off the end of your own automation treadmill?
+
+Here are a few hard-learned principles that I draw upon:
+
+1. **Shift down** — tackle failure as close as possible to the failure source to limit the risk of injuring yourself with cascading failures.
+2. **Build every layer** — and  every system,  so that it is built to fail.
+3. ***Build***** every layer** — do not think you have DR when you have HA; and don’t think you have HA just because you have one workload that spans several clusters.
```

Comment:
```suggestion
3. ***Build*** *every layer* — do not think you have DR when you have HA, and don’t think you have HA just because you have one workload that spans several clusters.
```